include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -428,7 +428,7 @@ __pattern_min_element(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __
     using _IteratorValueType = typename ::std::iterator_traits<_Iterator>::value_type;
     using _IndexValueType = ::std::make_unsigned_t<typename ::std::iterator_traits<_Iterator>::difference_type>;
     using _ReduceValueType = tuple<_IndexValueType, _IteratorValueType>;
-#ifdef _ONEDPL_DETECT_SPIRV_COMPILATION
+#if defined(_ONEDPL_DETECT_SPIRV_COMPILATION) && _ONEDPL_DETECT_SPIRV_COMPILATION
```

Comment:
Why do we check if it is defined in this situation, but not in the other?
Looking at `_ONEDPL_SYCL_USM_HOST_PRESENT` and `_ONEDPL_LIBSYCL_VERSION` as an example, it seems that it is reasonable to not check if it is defined before referencing it directly.

Reply:
I added this check here since in this file we are not guaranteed to be using the SYCL backend whereas we are in `unseq_backend_sycl.h`.

However, it is okay to just check `_ONEDPL_DETECT_SPIRV_COMPILATION` directly as it will evaluate to 0 if undefined. I will make this change.

Reply:
Aha, fair enough.  Yes, I believe the C++ standard states explicitly that these undefined PP identifiers are replaced by `0`.